### PR TITLE
Fix handling of empty strings for MySQL SSL attributes and that values considered empty() also disable server cert verification

### LINKED ISSUE #42: MySQL adapter should not accept empty strings for the SSL attributes
This leads to errors because empty strings are valid, but useless values for the driver.

### PR DESCRIPTION
**Pull request by lippserd**
fixes #42

### CODE REVIEW

File: src/Adapter/Mysql.php [hunk 1]
```diff
@@ -23,30 +23,30 @@ public function getOptions(Config $config)
     {
         $options = parent::getOptions($config);
 
-        if (isset($config->use_ssl) && $config->use_ssl === '1') {
-            if (isset($config->ssl_key)) {
+        if (isset($config->use_ssL) && $config->use_ssl) {
```
Comment (nilmerg) [inline src/Adapter/Mysql.php:26]: You've got a typo there.

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: src/Adapter/Mysql.php (hunk 1 above)
Comment (lippserd) [reply src/Adapter/Mysql.php:26]: Whoops, how did that happen. Sry!

File: src/Adapter/Mysql.php [hunk 2]
```diff
@@ -23,30 +23,30 @@ public function getOptions(Config $config)
     {
         $options = parent::getOptions($config);
 
-        if (isset($config->use_ssl) && $config->use_ssl === '1') {
-            if (isset($config->ssl_key)) {
+        if (isset($config->use_ssl) && $config->use_ssl) {
```
Comment (nilmerg) [inline src/Adapter/Mysql.php:26]: And what happens here if `use_ssl = "0"`?

Comment (lippserd) [reply src/Adapter/Mysql.php:26]: `"0"` evaluates to false and `"1"` evaluates to true. I've removed the strict check to also take real boolean values into account. If we also want to account `"false"` and `"true"` we need to use `filter_var()`. Do you want that too?

Comment (nilmerg) [reply src/Adapter/Mysql.php:26]: No, I just wondered why you're not using `empty` here as well.

Comment (lippserd) [reply src/Adapter/Mysql.php:26]: Yeah, that makes me wonder now too 😆

**Review by nilmerg — APPROVED** [review]

**Pull request merged**